execution_engine/src/core/runtime/externals.rs

```diff
@@ -554,12 +555,17 @@ where
             FunctionIndex::CreateContractPackageAtHash => {
                 // args(0) = pointer to wasm memory where to write 32-byte Hash address
                 // args(1) = pointer to wasm memory where to write 32-byte access key address
-                let (hash_dest_ptr, access_dest_ptr) = Args::parse(args)?;
+                // args(2) = boolean flag to determine if the contract can be versioned
+                let (hash_dest_ptr, access_dest_ptr, is_locked) = Args::parse(args)?;
                 self.charge_host_function_call(
                     &host_function_costs.create_contract_package_at_hash,
                     [hash_dest_ptr, access_dest_ptr],
                 )?;
-                let (hash_addr, access_addr) = self.create_contract_package_at_hash()?;
+                let (hash_addr, access_addr) = if is_locked {
+                    self.create_contract_package_at_hash(ContractPackageStatus::Locked)?
```

Comment:
:+1: But why not simplify further with

```
impl From<bool> for ContractPackageStatus
```

or just add `ContractPackageStatus::new` so following is possible:

```
let package_status = ContractPackageStatus::new(is_locked);
self.create_contract_package_at_hash(package_status);
```

Reply:
Added a `ContractPackageStatus::new` to help code simplification